modules/packages/LinearAlgebra.chpl

```diff
@@ -340,64 +342,72 @@ class ExpmPadeHelper {
     return this.exactD10;
   }
 
-  proc comp_looseD4(){
-    if this.useExactOneNorm then{
+  proc comp_looseD4() {
+    if this.useExactOneNorm || isComplexType(this.eltType) then {
       return comp_exactD4();
     }
     else {
       if !isApproxdComputed["A4"] then {
-        //TODO: Need to write a function to compute the
-        // estimated norm (change the below function)
-        // Right now, just returns the exact norm
-        this.approxD4 = norm(comp_A4(), normType.norm1)**(1/4.0);
+        try {
+          this.approxD4 = onenormest(comp_A4())**(1/4.0);
+        }
+        catch e{
```

Comment:
It might be a better style to change this try/catch into an if/else where you check for squareness of the matrix (I believe that's the condition where `onenormest` could throw)